File: HYPERLEDGER/bfn-network-chaincode/bfnwebapi-node/src/server/models/auto-trade-order.ts
import { JsonObject, JsonProperty } from 'json2typescript';

@JsonObject('AutoTradeOrder')
export class AutoTradeOrder {
  @JsonProperty('docType', String)
  public docType?: string;
  @JsonProperty('name', String)
  public name: string | undefined;
  @JsonProperty('autoTradeOrderId', String)
  public autoTradeOrderId: string | undefined;
  @JsonProperty('date', String)
  public date: string | undefined;
  @JsonProperty('investorName', String)
  public investorName: string | undefined;
  @JsonProperty('dateCancelled', String)
  public dateCancelled: string | undefined;
  @JsonProperty('wallet', String)
  public wallet: string | undefined;
  @JsonProperty('investorProfile', String)
  public investorProfile: string | undefined;
  @JsonProperty('user', String)
  public user: string | undefined;
  @JsonProperty('investor', String)
  public investor: string | undefined;
  @JsonProperty('isCancelled', Boolean)
  public isCancelled: boolean | undefined;
}

File: HYPERLEDGER/bfn-network-apps/businesslibrary/lib/data/auto_trade_order.dart
class AutoTradeOrder {
  String autoTradeOrderId;
  String date;
  String investorName;
  String dateCancelled, wallet;
  String investorProfile, user, investor;
  bool isCancelled;
  AutoTradeOrder(
      {this.autoTradeOrderId,
      this.date,
      this.investor,
      this.investorName,
      this.dateCancelled,
      this.isCancelled,
      this.wallet,
      this.investorProfile,
      this.user});

  AutoTradeOrder.fromJson(Map data) {
    this.autoTradeOrderId = data['autoTradeOrderId'];
    this.date = data['date'];
    this.investorName = data['investorName'];
    this.dateCancelled = data['dateCancelled'];
    this.investorProfile = data['investorProfile'];
    this.user = data['user'];
    this.investor = data['investor'];
    this.isCancelled = data['isCancelled'];
    this.wallet = data['wallet'];
  }
  Map<String, dynamic> toJson() => <String, dynamic>{
        'autoTradeOrderId': autoTradeOrderId,
        'date': date,
        'investorName': investorName,
        'dateCancelled': dateCancelled,
        'investorProfile': investorProfile,
        'user': user,
        'investor': investor,
        'isCancelled': isCancelled,
        'wallet': wallet,
      };
}
